packages/vats/test/test-lib-chainPublish.js

```diff
@@ -0,0 +1,166 @@
+// @ts-check
+import { test } from '@agoric/swingset-vat/tools/prepare-test-env-ava.js';
+
+import { E } from '@endo/far';
+import { publishToChainNode } from '../src/lib-chainPublish.js';
+
+test('publishToChainNode', async t => {
+  // eslint-disable-next-line no-use-before-define
+  const inputs = getMockInputs(t);
+  const {
+    // external facets
+    storageNode,
+    timerService,
+    customAsyncIterable,
+    // internal facets
+    nodeValueHistory,
+    advanceTimer,
+    supplyIterationResult,
+  } = inputs;
+  publishToChainNode(customAsyncIterable, storageNode, { timerService });
+
+  supplyIterationResult('foo');
+  await E(Promise).resolve();
```

Comment:
:exploding_head:

Reply:
Yeah, I needed a sufficient delay to match the code under test and didn't like the approach of appending `.then()`s to something like `Promise.resolve()` until I observed progress.